Assets/Scripts/GuiOutput.cs

```diff
@@ -0,0 +1,92 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+using System.Collections.Generic;
+
+public class GuiOutput : MonoBehaviour {
+
+    static private Text m_infoMessage;
+    static private Text m_contextMessage;
+    static private Text m_debugDistanceMessage;
+
+    public readonly float MAX_DELTA = 5.0F;
+    static private float elapsed;
+
+    static private Queue<string> m_messages;
+
+    private void Start()
+    {
+        m_debugDistanceMessage = GameObject.FindGameObjectWithTag("DebugDistance").GetComponent<Text>() as Text;
+        m_infoMessage = GameObject.FindGameObjectWithTag("InfoMessage").GetComponent<Text>() as Text;
+        m_contextMessage = GameObject.FindGameObjectWithTag("ContextMessage").GetComponent<Text>() as Text;
+        m_messages = new Queue<string>();
+        elapsed = 0.0F;
+    }
+
+    private void Update()
+    {
+
+        if (m_infoMessage == null) { return; }
+
+        if (m_infoMessage.text != "")
+        {
+            elapsed += Time.deltaTime;
+        }
+        else
+        {
+            DisplayNextInfoMessage();
+        }
+
+        if (elapsed > MAX_DELTA)
+        {
+            DisplayNextInfoMessage();
+            elapsed = 0.0F;
+        }
+    }
+
+    void DisplayNextInfoMessage()
+    {
+
+        if (m_infoMessage == null) { return; }        
+

```

Comment:
If the GUICanvas is not put into the prefab (or more accurately, if there are no Texts with labels InfoMessage, DebugDistance, ContextMessage) then they will be null.